frontend/src/util/fetchActionPlan.ts

```
@@ -93,9 +97,22 @@ export async function fetchActionPlan(
     const actionPlan = JSON.parse(fixedJson);
     /* eslint-enable */
 
-    return actionPlan as ActionPlan;
+    return { actionPlan: actionPlan as ActionPlan };
   } catch (error) {
+    // Check if the error is due to timeout
+    if (error instanceof Error && error.name === "AbortError") {
+      return {
+        actionPlan: null,
+        errorMessage: "Request timed out, please try again.",
```

Should this `throw` an error with the `errorMessage` rather than return the `errorMessage`?

---

As I understand it, and tested locally, throwing an error causes rendering issues. The fetch function throws an error and we catch it within our try catch block, but we want our broader fetchResources and fetchActionPlan functions to return content that can be rendered on screen.

---

> throwing an error causes rendering issues

Curious. 

> we want our broader fetchResources and fetchActionPlan functions to return content that can be rendered on screen

They can raise specific [custom errors](https://javascript.info/custom-errors) that can be caught and the message in the error can then be rendered however we like.

Perhaps my idea is over-engineered for a pilot.